contracts/RouterV2.sol

```diff
@@ -18,6 +15,11 @@ interface iPOOLFACTORY {
 interface iLEND {
     function checkInterest(address) external ;
 }
+interface iRESERVE {
+    function grantFunds(uint, address) external returns(bool); 
+}
```

Comment:
Add grantFunds function to pull the dividend from the Reserve on valid 'addDividend' call